internal/impl/config.go

```diff
@@ -101,6 +102,20 @@ func (c *Config) InitHook() *shellcmd.Commands {
 	return c.Shell.InitHook
 }
 
+func (c *Config) Merge(other *Config) {
+	if other == nil {
+		return
+	}
+	c.Packages = lo.Uniq(append(c.Packages, other.Packages...))
+	c.Env = lo.Assign(c.Env, other.Env)
+	c.InitHook().Merge(other.InitHook())
+	if c.Shell == nil {
+		c.Shell = &shellConfig{}
+	}
+	c.Shell.Scripts = lo.Assign(c.Scripts(), other.Scripts())
+	c.Include = append(c.Include, other.Include...)
```

Comment:
Can you add a comment near `Config` struct fields that we need to update this function when adding a new field?

Bonus points for adding a testcase with two Config structs to be merged, which asserts:
1. both structs have all non-zero-value fields. This ensures that new fields get added to the test case
2. the merged struct has fields that are different from either source struct's corresponding field. This ensures this Merge function was updated